src/Tasks/LegalProposition/Quiz.hs

```diff
@@ -43,11 +43,12 @@ generateLegalPropositionInst LegalPropositionConfig  {syntaxTreeConfig = SynTree
     pseudoFormulas <- genPseudoList serialsOfWrong serialsOfBracket treeList `suchThat` noSimilarFormulas
     return $ LegalPropositionInst
         { serialsOfWrong = fromList serialsOfWrong
-        , pseudoFormulas = pseudoFormulas
+        , pseudoFormulas = map replaceUnderscores pseudoFormulas
         , correctTrees = [ tree | (index, tree) <- zip [1..] treeList, index `notElem` serialsOfWrong ]
         , showSolution = printSolution
         , addText = extraText
         }
+          where replaceUnderscores = DLE.replace "_" "" . DLE.replace "_ " "" . DLE.replace " _" ""
```

Comment:
Can you try (after putting it into `illegalDisplay`) whether replacing lone `_` is really necessary?

Reply:
This is necessary. Otherwise, something like this could occur `¬B ∨ ((B ∧ ¬_) ∨ ¬C)`